example/lib/BUILD.bazel

```diff
@@ -1,10 +1,11 @@
-load("@aspect_bazel_lib//lib:copy_to_directory.bzl", "copy_to_directory")
+load("//js:js_package.bzl", "js_package")
 
-copy_to_directory(
+js_package(
     name = "lib",
     srcs = [
         "index.js",
         "package.json",
     ],
+    package = "@mycorp/mylib",
```

Comment:
ideally the fact that package.json is in srcs would be enough for us to read this, but I guess you want to put it in the provider which means you can't use an action to read the value :(

Reply:
yeah; no way to read it out of the package.json in a rule to get it into a provider

Reply:
the other direction of generating a default package.json if the user hasn't provided one we can tho however, which may be the more common pattern

Reply:
If both exist like in this case will or should we set the package name within the package.json?

Reply:
In theory we could have the validation action spit out a whole new tree artifact with the mutated package json. Tho that behaviour seems surprising. Instead I think we should auto-populate the package with the gazelle plug-in. Verify action should run as well similar to how we verify tsconfig settings in ts_project